src/unit/test_bitops.c

```diff
@@ -0,0 +1,80 @@
+#include <time.h>
+#include <stdint.h>
+
+#include "test_help.h"
+
+#include "../config.h"
+#include "../zmalloc.h"
+
+extern long long popcountScalar(void *s, long count);
+#ifdef HAVE_AVX2
+extern long long popcountAVX2(void *s, long count);
+#endif
+
+static long long bitcount(void *s, long count) {
+    long long bits = 0;
+    uint8_t *p = (uint8_t *)s;
+    for (int x = 0; x < count; x += 1) {
+        uint8_t val = *(x + p);
+        while (val) {
+            bits += val & 1;
+            val >>= 1;
+        }
+    }
+    return bits;
+}
+
+static int test_case(const char *msg, int size) {
+    uint8_t buf[size];
+    int fuzzing = 1000;
+    for (int y = 0; y < fuzzing; y += 1) {
+        for (int z = 0; z < size; z += 1) {
+            buf[z] = random() % 256;
+        }
+
+        long long expect = bitcount(buf, size);
+        long long ret_scalar = popcountScalar(buf, size);
+        TEST_ASSERT_MESSAGE(msg, expect == ret_scalar);
+#ifdef HAVE_AVX2
+        long long ret_avx2 = popcountAVX2(buf, size);
+        TEST_ASSERT_MESSAGE(msg, expect == ret_avx2);
+#endif
+    }
+
+    return 0;
+}
+
+int test_popcount(int argc, char **argv, int flags) {
+    UNUSED(argc);
+    UNUSED(argv);
+    UNUSED(flags);
+
+    time_t seed = time(NULL);
+    srandom(seed);
+    TEST_PRINT_INFO("Verify the popcount through fuzz testing,"
+                    " the random number seed is %jd.",
+                    (intmax_t)seed);
```

Comment:
I just merged #1751 which changed the unit test framework to seed the random generator before each test suite and print the seed, and it's possible to reproduce using a new `--seed` command line option. So I think we should not seed it here.